Sources/SPMTestSupport/PackageGraphTester.swift

```diff
@@ -179,13 +179,17 @@ public final class ResolvedTargetResult {
     }
 
     public func checkDerivedPlatforms(_ platforms: [String: String], file: StaticString = #file, line: UInt = #line) {
-        let targetPlatforms = Dictionary(uniqueKeysWithValues: target.platforms.derived.map({ ($0.platform.name, $0.version.versionString) }))
+        let derived = platforms.map {
+            let platform = PlatformRegistry.default.platformByName[$0.key] ?? PackageModel.Platform.custom(name: $0.key, oldestSupportedVersion: $0.value)
+            return target.platforms.getDerived(for: platform, usingXCTest: target.type == .test)
+        }
+        let targetPlatforms = Dictionary(uniqueKeysWithValues: derived.map({ ($0.platform.name, $0.version.versionString) }))
         XCTAssertEqual(platforms, targetPlatforms, file: file, line: line)
     }
```

Comment:
nit: formatting

Reply:
I... guess?

`swiftformat` comes up with some utterly bizarre stuff here:

```
let targetPlatforms = Dictionary(
            uniqueKeysWithValues: derived
                .map { ($0.platform.name, $0.version.versionString) }
        )
```

but I guess we like it...

Reply:
@MaxDesiatov do we?

Reply:
IIUC if you put `.map { ($0.platform.name, $0.version.versionString) }` on the previous line it won't break it up again, since that wouldn't trigger line length checks anymore

Reply:
Maybe, I just think a line length check is wrong to begin with and this is just one example where it produces something that's less readable than the original.